File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for s in tokens:
            if s == "+":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(val2 + val1)
            elif s == "-":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(val2 - val1)
            elif s == "*":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(val2 * val1)
            elif s == "/":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(int(val2 / val1))
            else:
                stack.append(int(s))
            print(stack)

        return stack.pop()

File: Data Structures & Algorithms/validate-parentheses/submission-2.java
class Solution {
    public boolean isValid(String s) {
         Stack<Character> stack = new Stack<>();

         for(int i = 0; i < s.length(); i++)    {
            Character c = s.charAt(i);
            if(c == ')')    {
                if(stack.isEmpty() || stack.pop() != '(') return false;
            } else if(c == '}')    {
                if(stack.isEmpty() || stack.pop() != '{') return false;
            } else if(c == ']')    {
                if(stack.isEmpty() || stack.pop() != '[') return false;
            } else  {
                stack.push(c);
            }

         }

        if(stack.isEmpty()) return true;
        return false;
         
    }
}
